packages/web-app-files/src/components/SideBar/SideBar.vue

```diff
@@ -1,198 +1,175 @@
 <template>
-  <oc-app-side-bar
-    :key="highlightedFile.id"
+  <div
     v-click-outside="onClickOutside"
-    class="files-sidebar oc-px-s oc-border-l"
-    :disable-action="false"
-    :close-button-label="$gettext('Close file sidebar')"
-    @close="close()"
+    :class="{
+      'has-active': !!appSidebarActivePanel
+    }"
   >
-    <template v-if="highlightedFile" slot="title">
-      <div class="uk-inline">
-        <oc-icon :name="highlightedFile.icon" size="xlarge" />
-      </div>
-      <div class="uk-inline">
-        <div class="uk-flex uk-flex-middle">
-          <h2
-            id="files-sidebar-item-name"
-            class="oc-text-initial oc-mr-s oc-text-bold uk-margin-remove"
-            tabindex="-1"
-            v-text="highlightedFile.name"
-          />
-        </div>
-        <div class="uk-flex uk-flex-middle">
-          <oc-button
-            v-if="!publicPage() && isFavoritesEnabled"
-            id="files-sidebar-star-icon"
-            :aria-label="
-              highlightedFile.starred
-                ? $gettext('Click to remove this file from your favorites')
-                : $gettext('Click to mark this file as favorite')
-            "
-            appearance="raw"
-            class="oc-mr-xs"
-            @click.native.stop="toggleFileFavorite(highlightedFile)"
-          >
-            <oc-icon :class="favoriteIconClass" name="star" />
-          </oc-button>
-          <template v-if="highlightedFile.size > -1">
-            {{ getResourceSize(highlightedFile.size) }},
-          </template>
-          {{ modificationTime }}
+    <div
+      v-for="panelMeta in panelMetas"
+      :id="`sidebar-panel-${panelMeta.app}`"
+      :key="`panel-${panelMeta.app}`"
+      ref="panels"
+      :tabindex="appSidebarActivePanel === panelMeta.app ? -1 : false"
+      class="sidebar-panel"
+      :class="{
+        'is-active':
+          appSidebarActivePanel === panelMeta.app || (!appSidebarActivePanel && panelMeta.default),
+        'sidebar-panel--default': panelMeta.default
+      }"
+    >
+      <div class="sidebar-panel__header header">
+        <oc-button
+          v-if="!panelMeta.default"
+          class="header__back"
+          appearance="raw"
+          :aria-label="$gettext('Back to previous sidebar')"
```

Comment:
As the aria-label overrides the button text for screen readers, it must (!) contain the visible button label. Meaning: You need to interpolate the parent panel title into the aria-label. Something like e.g. `Back to Details panel`.

Reply:
Fixed it by including the name of the previous panel. Also cleaned up the `reduce` for collecting panels and the sidebar definition a bit.